test/analyzer_test.rb

```diff
@@ -1,16 +1,22 @@
 require "test_helper"
 
 class AnalyzerTest < Minitest::Test
-  def test_that_it_converts_to_path_and_runs
-    slug = mock
-    solution_path = mock
-    output_path = mock
-    solution_pathname = mock
-    output_pathname = mock
-    Pathname.expects(:new).with(solution_path).returns(solution_pathname)
-    Pathname.expects(:new).with(output_path).returns(output_pathname)
+  TMP_PATH = Pathname.new('/tmp')
 
-    AnalyzeSolution.expects(:call).with(slug, solution_pathname, output_pathname)
-    Analyzer.analyze(slug, solution_path, output_path)
+  protected
+  attr_reader :exercise_slug
+
+  def analysis_results(code = "")
+    solution_path = "#{TMP_PATH}/#{FILENAMES[exercise_slug]}"
+    analysis_path = "#{TMP_PATH}/analysis.json"
+    File.open(solution_path, "w") do |f|
+      f.write("#{code}\n")
+    end
+    AnalyzeSolution.(exercise_slug, TMP_PATH, TMP_PATH)
+    file = File.open(analysis_path)
+    JSON.load(file)
```

Comment:
If you mean to do this, feel free to tell Rubocop and us that you do.

You can given an [inline directive] if wanted.

[inline directive]: https://stackoverflow.com/a/26350879/485864

Reply:
Thanks, the tests filed when I made the change that Rubocop requested so I added an inline directive as you suggested. This is something we might want to change in the rubocop config.